lib/rando.js

```diff
@@ -8,60 +8,76 @@ var Rando = mongoose.model("rando", new mongoose.Schema({
     creation: Number,
     ip: String,
     location: {
-    latitude: Number,
-    longitude: Number
+      latitude: Number,
+      longitude: Number
     },
     imageURL: String,
     imageSizeURL: {
-    small: String,
-    medium: String,
-    large: String
+      small: String,
+      medium: String,
+      large: String
     },
     mapURL: String,
     mapSizeURL: {
-    small: String,
-    medium: String,
-    large: String
+      small: String,
+      medium: String,
+      large: String
     },
     rating: Number,
     delete: Number
 }));
 
 module.exports = {
     add: function (rando, callback) {
-    winston.info("DB.Rando: Add rando by: ", rando.email);
-    new Rando(rando).save(callback);
+      winston.info("DB.Rando: Add rando by: ", rando.email);
+      new Rando(rando).save(callback);
+    },
+    getByRandoId: function (randoId, callback) {
+      winston.info("DB.Rando: get by randoId:", randoId);
+      Rando.findOne({randoId: randoId}, callback);
     },
     getAll: function (callback) {
-    winston.info("DB.Rando: get all");
-    Rando.find({}, callback);
+      winston.info("DB.Rando: get all");
+      Rando.find({}, callback);
     },
     getFirstN: function (N, callback) {
-    winston.info("DB.Rando: get first " + N);
-    Rando.find({}).limit(N).exec(callback);
+      winston.info("DB.Rando: get first ", N);
+      Rando.find({}).limit(N).exec(callback);
     },
     remove: function (rando, callback) {
-    winston.info("Db.Rando: Remove rando: ", rando.email, " randoId: ", rando.randoId);
-    rando.remove(function (err) {
+      winston.info("Db.Rando: Remove rando: ", rando.email, " randoId: ", rando.randoId);
+      rando.remove(function (err) {
+        if (err) {
+          winston.warn("DB.Rando: Can't remove rando: " , rando.randoId, " because: ", err);
+        }
+
+        if (callback) {
+          callback(err);
+        }
+      });
+    },
+    removeById: function (randoId, callback) {
+      var self = this;
```

Comment:
I think we need to implement this method like this:
Rando.remove({randoId: randoId}, callback);